tests/failpoints/cases/test_cmd_epoch_checker.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_rocks::RocksSnapshot;
+use kvproto::raft_cmdpb::{RaftCmdRequest, RaftCmdResponse};
+use raft::eraftpb::MessageType;
+use raftstore::store::msg::*;
+use test_raftstore::*;
+use tikv_util::HandyRwLock;
+
+use std::sync::mpsc::{self, TryRecvError};
+use std::time::Duration;
+
+struct CbReceivers {
+    proposed: mpsc::Receiver<()>,
+    committed: mpsc::Receiver<()>,
+    applied: mpsc::Receiver<RaftCmdResponse>,
+}
+
+impl CbReceivers {
+    fn assert_not_ready(&self) {
+        self.proposed
+            .recv_timeout(Duration::from_millis(100))
+            .unwrap_err();
+        self.committed
+            .recv_timeout(Duration::from_millis(100))
+            .unwrap_err();
+        self.applied
+            .recv_timeout(Duration::from_millis(100))
+            .unwrap_err();
+    }
+
+    fn assert_ok(&self) {
+        let resp = self
+            .applied
+            .recv_timeout(Duration::from_millis(100))
+            .unwrap();
+        assert!(
+            !resp.get_header().has_error(),
+            "{:?}",
+            resp.get_header().get_error()
+        );
+        // proposed and committed should be invoked before applied
+        self.proposed.try_recv().unwrap();
```

Comment:
Actually, the `proposed` and `committed` may not be invoked if this proposal is proposed when the leader has not applied to current term.
So before this proposal which needs to call `assert_ok`, we should ensure the leader has applied to current term by proposing any proposal or waiting a while. I prefer the former way.